roads.mss

```diff
@@ -2667,6 +2667,428 @@ come in as well.
   }
 }
 
+#roads_high::surface[zoom>=16],
+#tunnel::surface[zoom>=16],
+#bridge::surface[zoom>=16] {
+  // Don't draw something for good surface, ok for racing bikes (<=28mm tyres).
+
+  // Surface ok for treking/gravel/cyclocross/city bike (28mm < tyres <= 40mm).
+  [surface='concrete:lanes'],
+  [surface='concrete:plates'],
+  [surface='paving_stones'],
+  [surface='sett'],
+  [surface='unhewn_cobblestone'],
+  [surface='cobblestone'],
+  [surface='wood'],
+  [surface='compacted'],
+  [surface='fine_gravel'],
+  [surface='woodchips'],
+  [tracktype='grade2'],
+  [tracktype='grade3'],
+  [smoothness='intermediate'],
+  [smoothness='bad'] {
+    [zoom=16] {
+      line-dasharray: 1,4;
+    }
+    [zoom=17] {
+      line-dasharray: 2,8;
+    }
+    [zoom>=18] {
+      line-dasharray: 4,16;
+    }
+
+    /* -- colors & styles -- */
+    line-color: darken(@standard-fill,@surfaceLighterDarker1);//Darken since its white.
+    [type='motorway'],
+    [type='motorway_link'] {
+      line-color: lighten(@motorway-fill,@surfaceLighterDarker1);
+    }
+    [type='trunk'],
+    [type='trunk_link'] {
+      line-color: lighten(@trunk-fill,@surfaceLighterDarker1);
+    }
+    [type='primary'],
+    [type='primary_link'] {
+      line-color: lighten(@primary-fill,@surfaceLighterDarker1);
+    }
+    [type='secondary'],
+    [type='secondary_link'] {
+      line-color: lighten(@secondary-fill,@surfaceLighterDarker1);
+    }
+    [type='track'] {
+        line-color: lighten(@track-fill, @surfaceLighterDarker1);
+    }
+    [type='cycleway'] {
+      line-color: lighten(@cycle-fill,@surfaceLighterDarker1);
+    }
+    [type='pedestrian'] {
+      line-color: lighten(@pedestrian-fill,@surfaceLighterDarker1);
+    }
+    [type='bridleway'] {
+      line-color: lighten(@bridleway-fill,@surfaceLighterDarker1);
+    }
+    [type='footway'] {
+      line-color: lighten(@footway-fill,@surfaceLighterDarker1);
+    }
+    [type='path'],
+    [type='footway'][bicycle='yes'],
+    [type='footway'][bicycle='designated'],
+    [type='bridleway'][bicycle='yes'],
+    [type='bridleway'][bicycle='designated'] {
+      line-color: lighten(@path-fill,@surfaceLighterDarker1);
+    }
+    [type='path'][access='no'][bicycle=null],
+    [type='path'][bicycle='no'] {
+      line-color: lighten(@path-nobicycle,@surfaceLighterDarker1);
+    }
+    [type='service'],
+    [type='track'],
+    [type='tertiary'],
+    [type='living_street'],
+    [type='road'],
+    [type='unclassified'],
+    [type='residential'],
+    [type='tertiary_link'],
+    [type='secondary_link'],
+    [type='primary_link'],
+    [type='trunk_link'],
+    [type='motorway_link'],
+    [type='primary'],
+    [type='secondary'],
+    [type='motorway'],
+    [type='trunk'] {
+      line-cap: round;
```

Comment:
Not sure we need the line-cap / line-join here (did not check in details). If this is useless, would probably ease a bit the filtering and render.

Reply:
yep